controllers/control/cassandratask_controller.go

```diff
@@ -209,22 +255,54 @@ func (r *CassandraTaskReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	var err error
 	var failed, completed int
 	for _, job := range cassTask.Spec.Jobs {
-		taskConfigProto := &TaskConfiguration{
+		taskConfig := &TaskConfiguration{
 			RestartPolicy: cassTask.Spec.RestartPolicy,
 			Id:            taskId,
+			Datacenter:    dc,
+			TaskStartTime: cassTask.Status.StartTime,
+			Context:       ctx,
+			Arguments:     job.Arguments,
 		}
+
+		// Process all the reconcileEveryPodTasks
 		switch job.Command {
-		case "rebuild":
-			res, failed, completed, err = r.reconcileEveryPodTask(ctx, &dc, rebuild(taskConfigProto, job.Arguments))
-		case "cleanup":
-			res, failed, completed, err = r.reconcileEveryPodTask(ctx, &dc, cleanup(taskConfigProto))
-		case "decommission":
-			// res, err = r.reconcileEveryPodTask(ctx, taskId, &dc, decommission())
+		case api.CommandRebuild:
+			rebuild(taskConfig)
+		case api.CommandCleanup:
+			cleanup(taskConfig)
+		case api.CommandRestart:
+			r.restart(taskConfig)
+		case api.CommandReplaceNode:
+			/*
+				TODO The death of those pods should cause a restart.. what happens if someone wants to replace all the pods? Will we kill
+				  the datacenter? Do we need to wait for the new pods to be alive before we continue? Can we requeue from the process
+				  to make sure we retry the process (it isn't finished yet..) or just make it stuck?
+			*/
+			r.replace(taskConfig)
+		case "forceupgraderacks":
+			// res, failed, completed, err = r.reconcileDatacenter(ctx, &dc, forceupgrade(taskConfigProto))
+		case api.CommandUpgradeSSTables:
+			upgradesstables(taskConfig)
+		case api.CommandScrub:
+			// res, failed, completed, err = r.reconcileEveryPodTask(ctx, &dc, scrub(taskConfigProto))
+		case api.CommandCompaction:
+			// res, failed, completed, err = r.reconcileEveryPodTask(ctx, &dc, compact(taskConfigProto, job.Arguments))
 		default:
 			err = fmt.Errorf("unknown job command: %s", job.Command)
 			return ctrl.Result{}, err
 		}
 
+		if err := taskConfig.Validate(); err != nil {
+			// TODO Or should we mark this task as failed and simply stop processing?
```

Comment:
The validation could fail e.g. if the pod name does not exist. If the user is able to rectify a wrong task argument, I think it makes sense to requeue, no?

Reply:
Yes, that's what it does now (allows fixing). But on the other hand, the "Job" type in Kubernetes just fails and refuses to rerun. So semantics are a bit different. That's why the question.